core/node/da_clients/src/avail/client.rs

```diff
@@ -42,15 +42,43 @@ pub struct AvailClient {
 pub struct BridgeAPIResponse {
     blob_root: Option<H256>,
     bridge_root: Option<H256>,
-    data_root_index: Option<u64>,
+    #[serde(deserialize_with = "deserialize_u256_from_integer")]
+    data_root_index: Option<U256>,
     data_root_proof: Option<Vec<H256>>,
     leaf: Option<H256>,
-    leaf_index: Option<u64>,
+    #[serde(deserialize_with = "deserialize_u256_from_integer")]
+    leaf_index: Option<U256>,
     leaf_proof: Option<Vec<H256>>,
     range_hash: Option<H256>,
     error: Option<String>,
 }
 
+#[derive(Deserialize)]
+#[serde(untagged)]
+enum U256Value {
+    Number(u64),
+    String(String),
+}
+
+fn deserialize_u256_from_integer<'de, D>(deserializer: D) -> Result<Option<U256>, D::Error>
+where
+    D: serde::Deserializer<'de>,
+{
+    use serde::de::Error;
+
+    match Option::<U256Value>::deserialize(deserializer)? {
+        Some(U256Value::Number(num)) => Ok(Some(U256::from(num))),
+        Some(U256Value::String(s)) => if let Some(stripped) = s.strip_prefix("0x") {
```

Comment:
Nit: A potentially stripped string can be simplified to `s.strip_prefix("0x").unwrap_or(&s)`.